imap/mailbox.c

```diff
@@ -3195,25 +3215,16 @@ static void header_update_counts(struct index_header *i,
     if (record->system_flags & FLAG_FLAGGED)
         i->flagged += num;
 
-    if (record->system_flags & FLAG_DELETED)
+    if (record->system_flags & FLAG_DELETED) {
         i->deleted += num;
+        UPDATE_QUOTA_USED(i->quota_deleted_used, record->size, is_add);
+    }
 
     if (!(record->system_flags & FLAG_SEEN))
         i->unseen += num;
 
-    if (is_add) {
-        i->exists++;
-        i->quota_mailbox_used += record->size;
-    }
-    else {
-        if (i->exists) i->exists--;
-
-        /* corruption prevention - check we don't go negative */
-        if (i->quota_mailbox_used > record->size)
-            i->quota_mailbox_used -= record->size;
-        else
-            i->quota_mailbox_used = 0;
-    }
+    i->exists += num;
```

Comment:
This loses the negative prevention of the old code for the `exists` field.

Reply:
Yep.  Implemented that check for all flag counts